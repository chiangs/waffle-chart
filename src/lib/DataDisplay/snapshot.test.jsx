import React from 'react';
import renderer from 'react-test-renderer';
import { describe, expect, test } from 'vitest';
import { DataDisplay } from './index';

describe('DataDisplay', () => {
    test('DataDisplay component renders correctly', () => {
        const props = {
            data1: {
                value: 1,
                count: 1,
                dataLabel: 'data1',
            },
            data2: {
                value: 2,
                count: 2,
                dataLabel: 'data2',
            },
            verticalFill: false,
        };
        const component = renderer.create(<DataDisplay {...props} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
