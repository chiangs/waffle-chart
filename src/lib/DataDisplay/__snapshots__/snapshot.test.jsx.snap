// Vitest Snapshot v1

exports[`DataDisplay > DataDisplay component renders correctly 1`] = `
<div
  className="waffle-chart-display-container false"
  data-testid="display-container"
>
  <div
    className="display-item"
    data-testid="display-item"
    style={
      {
        "color": "",
      }
    }
  >
    <h3
      className="display-item-title"
      data-testid="display-percentage"
    >
      1
      %
    </h3>
    <p
      className="display-item-description"
      data-testid="display-count"
    >
      1
       
      data1
    </p>
  </div>
  <div
    className="display-item"
    data-testid="display-item"
    style={
      {
        "color": "",
      }
    }
  >
    <h3
      className="display-item-title"
      data-testid="display-percentage"
    >
      2
      %
    </h3>
    <p
      className="display-item-description"
      data-testid="display-count"
    >
      2
       
      data2
    </p>
  </div>
</div>
`;
